### DISCUSSION TITLE
Select genres when playing tracks randomly

### DISCUSSION THREAD

**Discussion by BjornB-Git** [MoleMan1024/audiowagon#159] [Ideas]
Thanks a million for having developed this app! First time I encountered Replay Gain in a Car Audio system, great stuff!

I just bought a 2024 Volvo T8 XC60 Recharge, which I wouldn’t have done without the possibility to play my music from an USB stick.
Volvo should pay you commission!  :)

Being a music freak there are still features to wish for. 

Number 1, I would like to be able to select genres to include when playing tracks randomly. I have a lot of tracks tagged “Easy Listening”, and “Smooth Jazz” which would simplify travelling with my wife :). 

Another thing would be to have track artwork displayed on the app banner together with artist, track name and album name.

Thanks
Bjorn

**Comment by MoleMan1024** [comment]
> Number 1, I would like to be able to select genres to include when playing tracks randomly. I have a lot of tracks tagged “Easy Listening”, and “Smooth Jazz” which would simplify travelling with my wife :).

There is already a request for genres in issue #81 . The problem is that the default Android Automotive media UI only allows to show up to 4 main categories, which I already use for Tracks, Albums, Artists and Files. So there would be no nice way to access genres.

> Another thing would be to have track artwork displayed on the app banner together with artist, track name and album name.

Not quite sure what you mean by "app banner". I don't know about the Volvos but in my Polestar 2 on the main screen the album art will show like this (the screenshot is from Android emulator, but it works like that in real car also): 

![Screenshot_20241031_162753](https://github.com/user-attachments/assets/8e55c43e-be44-4ff9-b08f-849b1f3f9674)

If Volvo's don't have that, I can't change it.

**Comment by BjornB-Git** [comment]
![Running apps screen](https://github.com/user-attachments/assets/ce9ffaf2-71c1-471a-8aa1-5affec2d3246)
![Settings screen](https://github.com/user-attachments/assets/848c4600-75e2-45ff-9145-a7de0d092c9e)
Hi and thanks for your answer

I guess there is a framework that you must adhere to regarding the 4 tabs in the UI and other stuff.
However, in settings, I can select what content I would like each of the four tabs to show from a list. I guess that list could be longer than four items and that genres could be one of them. Please see attached image translated by google from Swedish.

Another thought for genres would be to have an expandable section in settings working as a filter for songs, with a checkbox for each genre and, of course, defaulting to “all genres”.

I have never developed for AAOS or any Android OS so, apologize if suggestions are totally off 😉


Regarding my second wish, yes noticed that most of it was already there. Just opacity is set so that the album art is hardly visible.
Image attached


Again, thanks a lot for your answer and for the APP!!!
Bjorn
